### PR TITLE
13.get/api

### PR DESCRIPTION
**Pull request by JadwigaPowroznik**

### CODE REVIEW

**Review by Angaca — CHANGES_REQUESTED** [review]
You could do some extra check that the specific comment has been deleted. I would query the db in your test directly to try to get what you deleted and expect empty rows in return. For the GET /api I would like to know how can I enquiry those endpoints; GET, PATCH, DELETE? not clear at the moment

File: __tests__/app.test.js [hunk 1]
```diff
@@ -459,4 +460,65 @@ describe("my Express app", () => {
...
   });
+  describe("/api/comments/:comment_id", () => {
+    it("204: responds with an empty response body and checks if comment has been deleted", () => {
+      const id = 8;
+      return request(app)
+        .delete(`/api/comments/${id}`)
+        .expect(204)
+        .then(({ body }) => {
+          expect(body).toEqual({});
+          db.query(`SELECT * FROM comments WHERE comment_id =${id}`).then(
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: I think you'd need a return before the db.query so the stack doesn't close without the async process continuing

File: controllers/news.controllers.js [hunk 2]
```diff
@@ -82,3 +84,35 @@ exports.postCommentByArticleId = async (req, res, next) => {
...
+          { endpoint: "/api/articles" },
+          { endpoint: "/api/articles/:article_id" },
+          { endpoint: "/api/articles/:article_id/comments" },
+          { endpoint: "/api/users" },
+        ],
+      },
+      { patch: [{ endpoint: "/api/articles/:article_id" }] },
+      { post: [{ endpoint: "/api/articles/:article_id/comments" }] },
+      { delete: [{ endpoint: "/api/comments/:comment_id" }] },
+    ],
```
Comment (Poonam-raj) [inline controllers/news.controllers.js]: What Andrea is mentioning with the endpoints is that as a user of the API I might go to '/api' to find out how to use the server. Just knowing the endpoints isn't enough in that case.

I might want to know what queries are available on an endpoint if any, what an example query looks like, what example responses from each endpoint might be. There's loads more detail we could add in. 

A great file to use as a reference is: https://github.com/northcoders/be-nc-news/blob/main/endpoints.json

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Still need some work on the `/api` endpoint, more detail needed. Other comment was addressed well :D

File: __tests__/app.test.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ const {
   articleData,
   commentData,
 } = require("../db/data/test-data/");
+const { expect } = require("@jest/globals");
```
Comment (emilyb93) [inline __tests__/app.test.js]: rogue import

File: __tests__/app.test.js [hunk 4]
```diff
@@ -459,4 +460,87 @@ describe("my Express app", () => {
...
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe("Incorrect data type passed to endpoint");
+        });
+    });
+  });
+  describe("/api", () => {
+    it("200: responds with a JSON describing all the available endpoints on this API", () => {
+      const endpointsJson = {
+        "GET /api": {
```
Comment (emilyb93) [inline __tests__/app.test.js:499]: You already have this somewhere else, just import the file and you can use it as an object

File: db/test.sql [hunk 5]
```diff
@@ -10,7 +10,10 @@
...
--- SELECT * FROM USERS;
+--  SELECT * FROM comments;
 -- INSERT INTO comments (author, body, article_id ) VALUES ('rogersop', 'aaa', 1) RETURNING comments.author AS username, comments.body
 
-SELECT articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, users.username AS author, COUNT(comment_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id=articles.article_id LEFT JOIN users ON users.username=articles.author WHERE articles.topic = 'mitch' GROUP BY articles.article_id, comments.article_id, users.username ORDER BY article_id DESC 
\ No newline at end of file
+-- SELECT articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, users.username AS author, COUNT(comment_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id=articles.article_id LEFT JOIN users ON users.username=articles.author WHERE articles.topic = 'mitch' GROUP BY articles.article_id, comments.article_id, users.username ORDER BY article_id DESC 
+
+DELETE FROM comments WHERE comment_id=8;
+SELECT * FROM comments WHERE comment_id =8
```
Comment (emilyb93) [inline db/test.sql:19]: Dont commit files that are not relevant to the commit

**Review by emilyb93 — CHANGES_REQUESTED** [review]
comments^

**Pull request merged**